### PR TITLE
Developer Documentation added

### PR DESCRIPTION
**Pull request by astitva1905**
Added developer documentation:
- Development workflow and intro in index.md
- Testing folder with documentation for tests
- architecture additions to system folder
- puml files convereted to mermaid

### CODE REVIEW

File: docs/developer/system/architecture.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+may lead to analytical insights.
+6. Save – save the state of a DT that’s already in the
+execution phase. This functionality is required for ondemand saving and re-spawning of DTs.
+7. What-if analysis – explore alternative scenarios to (i)
+plan for an optimal next step, (ii) recalibrate new DT
+assets, (iii) automated creation of new DTs or their
+assets; these newly created DT assets may be used to
+perform scientifically valid experiments.
+8. Share – share a DT with other users of their organisation.
+
```
Comment (prasadtalasila) [inline docs/developer/system/architecture.md:30]: The paper also contains a short description for each of the system components. Please add them after the diagram.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
Please see the changes made in [this branch](https://github.com/prasadtalasila/DTaaS-Public/tree/pr-114). You can merge those changes and resolve the comments given here.

There are some markdown linting problems. These are not being shown by codeclimate. Can you please check this? Thanks.

File: docs/developer/system/architecture.md (hunk 1 above)
Comment (astitva1905) [reply docs/developer/system/architecture.md:30]: The descriptions have been shifted to the respective pages of the components. Should I repeat them here as well?

File: docs/developer/system/lib-ms.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# Library Microservice
+
+[The Library Microservices](https://github.com/INTO-CPS-Association/DTaaS/tree/feature/distributed-demo/servers/lib#readme) - fulfil the core requirements of the system. The microservices are complementary and composable. 
+
+The service mesh enables discovery of microservices, load balancing and authentication functionalities. There are microservices for catering to author, store, explore, configure, execute and scenario analysis requirements.
+
+## Lib-MS Architecture
+
+### Class Architecture
+```mermaid
```
[the remark this answers is not in the document]
Comment (astitva1905) [reply docs/developer/system/lib-ms.md:10]: Updated, added configuration. Please have a look.

File: docs/developer/system/architecture.md (hunk 1 above)
Comment (prasadtalasila) [reply docs/developer/system/architecture.md:30]: We have documentation only for Client and Library microservice. Rest of the components are not in place. It makes sense to copy all of it here from the paper.

**Comment by prasadtalasila** [comment]
merged via #117

**Pull request closed without merging**